main.go

```diff
@@ -50,19 +55,40 @@ func main() {
 	}
 
 	jobQueue := filterJobQueue.CreateJobQueue(producer.Output())
-	router := mux.NewRouter()
 
-	s := server.New(cfg.BindAddr, router)
+	apiErrors := make(chan error)
```

Comment:
Same here -  if the creating the filter api sends a error to this chan before the select block is reached it will block.